WordPress/Classes/Utility/CoreDataManager.swift

```diff
@@ -0,0 +1,401 @@
+                return
+            }
+
+            updatedObjects.forEach {
+                do {
+                    let object = try context.existingObject(with: $0.objectID)
+                    if object.isFault {
+                        object.willAccessValue(forKey: nil)
+                    }
+                } catch {
+                    DDLogError("Error merging object \($0), \(error)")
+                }
+            }
+            context.mergeChanges(fromContextDidSave: notification)
+        }
+    }
+}
+
+
+// MARK: - Migration
+extension CoreDataManager {
+
+    private func migrateDataModelIfNecessary() {
+        guard FileManager.default.fileExists(atPath: storeURL.path) else {
+            DDLogInfo("No store exists at URL \(storeURL).  Skipping migration.")
+            return
+        }
+
+        guard let metadata = try? NSPersistentStoreCoordinator.metadataForPersistentStore(ofType: NSSQLiteStoreType, at: storeURL, options: nil) else {
+            return
+        }
+
+        guard managedObjectModel.isConfiguration(withName: nil, compatibleWithStoreMetadata: metadata) == false else {
+            // Configuration is compatible, no migration necessary.
+            return
+        }
+
+        DDLogWarn("⚠️ [CoreDataManager] Migration required for persistent store")
+
+        // Extract model names
+        let versionPath = modelURL.appendingPathComponent(Constants.versionInfoPlist).path
+        guard let versionInfo = NSDictionary(contentsOfFile: versionPath),
+            let modelNames = versionInfo[Constants.versionHashesKey] as? NSDictionary,
+            let allKeys = modelNames.allKeys as? [String],
+            let objectModel = NSManagedObjectModel(contentsOf: modelURL) else {
+                return
+        }
+
+        let sortedKeys = allKeys.sorted { (string1, string2) -> Bool in
+            return string1.compare(string2, options: [.numeric], range: nil, locale: nil) == .orderedAscending
+        }
+
+        do {
+            try CoreDataIterativeMigrator.iterativeMigrate(sourceStore: storeURL,
+                                                           storeType: NSSQLiteStoreType,
+                                                           to: objectModel,
+                                                           using: sortedKeys)
+        } catch {
+            DDLogError("☠️ [CoreDataManager] Unable to migrate store with error: \(error)")
+            sentryStartupError.add(error: error)
+        }
+    }
+}
+
+
+// MARK: - Notifications
+extension CoreDataManager {
+
+    private func observe() {
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.NSManagedObjectContextDidSave,
+                                               object: self.mainContext,
+                                               queue: nil) { [weak self] notification in
+            guard let self = self else {
```

Comment:
Should `self` be enclosed in `s?

Reply:
Since `Swift 4.2`, it is allowed to use the same keyword without back ticks
https://github.com/apple/swift-evolution/blob/master/proposals/0079-upgrade-self-from-weak-to-strong.md

Reply:
til!